File: typescipt/existing/src/language/expressions/functions/minuteFunction.ts



namespace Lexy.Compiler.Language.Expressions.Functions;

public class MinuteFunction : SingleArgumentFunction
{
   public const string Name = "MINUTE";

   protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

   protected override VariableType ArgumentType => PrimitiveType.Date;
   protected override VariableType ResultType => PrimitiveType.Number;

   private MinuteFunction(Expression valueExpression, SourceReference reference)
     : base(valueExpression, reference)
   {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression expression)
   {
     return new MinuteFunction(expression, reference);
   }
}

File: src/Lexy.Compiler/Language/Expressions/Functions/MinuteFunction.cs
using Lexy.Compiler.Language.VariableTypes;

namespace Lexy.Compiler.Language.Expressions.Functions;

public class MinuteFunction : SingleArgumentFunction
{
    public const string Name = "MINUTE";

    protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

    private MinuteFunction(Expression valueExpression, ExpressionSource source)
        : base(Name, valueExpression, source, PrimitiveType.Date, PrimitiveType.Number)
    {
    }

    public static FunctionCallExpression Create(ExpressionSource source, Expression expression)
    {
        return new MinuteFunction(expression, source);
    }
}